### ISSUE TITLE
pydroid3

### ISSUE THREAD

**Issue by pydroid888** [ranaroussi/yfinance#1419]
I'm running on pydroid3 and encountered missing "EPIC" error last week - has worked well for months. On attempting to upgrade yfinance i have repeated libxml2 & libxst dev pakage errors flagged. Previous installation were simple. These dependencies are missing and I can't install. Is this is a common issue? Is there a way yfinance can now allow for this - it has been so reliable for ages. Running pydroid3 on Samsung Galaxy Tablet - for ease of use anywhere.

/storage/emulated/0 $ pip install yfinance
Collecting yfinance
  Using cached yfinance-0.2.11-py2.py3-none-any.whl (59 kB)
Requirement already satisfied: appdirs>=1.4.4 in /data/data/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages (from yfinance) (1.4.4)
Requirement already satisfied: numpy>=1.16.5 in /data/data/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages (from yfinance) (1.21.2)
Collecting cryptography>=3.3.2
  Using cached cryptography-39.0.1.tar.gz (603 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: beautifulsoup4>=4.11.1 in /data/data/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages (from yfinance) (4.11.1)
Collecting html5lib>=1.1
  Using cached html5lib-1.1-py2.py3-none-any.whl (112 kB)
Requirement already satisfied: pandas>=1.3.0 in /data/data/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.9/site-packages (from yfinance) (1.3.3)
Collecting lxml>=4.9.1
  Using cached lxml-4.9.2.tar.gz (3.7 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [3 lines of output]
      Building lxml version 4.9.2.
      Building without Cython.
      Error: Please make sure the libxml2 and libxslt development packages are installed.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
/storage/emulated/0 $

**Comment by Gilles-Ain** [comment]
I have the same message error with pydroid3, no problème before!
It IS impossible toi installé libxml2 and libxlst.
No issue for mu ...

**Comment by pydroid888** [comment]
Libxml2 needs to be built, along with libxst, for pydroid, from source. It works fine on a Windows laptop. Pydroid dont have an updated lxml. How to build the necessary dependencies to update lxml?

**Comment by sidineicruz** [comment]
Estou com o mesmo problema. Como resolver?

**Comment by pydroid888** [comment]
No solution yet. Very annoying.

**Comment by paulmcq** [comment]
The error message
  `Error: Please make sure the libxml2 and libxslt development packages are installed.`
seems to say to install 'libxml2-dev' and 'libxslt-dev'.
On my Ubuntu Focal system I see 'libxml2-dev:amd64' and 'libxslt1-dev:amd64'.
HTH

**Comment by pydroid888** [comment]
You may be OK but both those packages are not on pydroid3 and need compiling too before lxml can be updated. Pydroid3 has become unusable with the newer yfinance. I don't know how to build the necessary packages/wheels to upgrade lxml to 4.9.2 or higher.

**Comment by pydroid888** [comment]
Have been unable to find anyone with the necessary knowledge or an existing solution.

**Comment by ValueRaider** [comment]
Have you tried simply not installing `lxml`, see what works/breaks? https://github.com/ranaroussi/yfinance/discussions/1080

**Comment by pydroid888** [comment]
Yes, yfinance and a couple of other modules won't work as they need the 4.9.2 or later version of lxml. Latest Web scraping tools.

**Comment by Gilles-Ain** [comment]
Yes it is a Big problem ...
And thé déveloper is russian!
WE have to wait a long Time to have a pydroid update maybe.


Le jeu. 13 avr. 2023 à 22:54, pydroid888 ***@***.***> a
écrit :

**Comment by pydroid888** [comment]
Have asked the developer, more than once, no responses.Sent from my Galaxy
-------- Original message --------From: Gilles-Ain ***@***.***> Date: 14/04/2023  13:26  (GMT+01:00) To: ranaroussi/yfinance ***@***.***> Cc: pydroid888 ***@***.***>, Author ***@***.***> Subject: Re: [ranaroussi/yfinance] pydroid3 (Issue #1419) 
Yes it is a Big problem ...
And thé déveloper is russian!
WE have to wait a long Time to have a pydroid update maybe.


Le jeu. 13 avr. 2023 à 22:54, pydroid888 ***@***.***> a
écrit :

**Comment by pydroid888** [comment]
Someone who knows how to compile the necessary modules for Android could sort this out. I dont, unfortunately. Compile from C with Android as the target. Lxml 4.9.2 and newer.

**Comment by pydroid888** [comment]
Does anyone know how to achieve this upgrade or can point me to a resource to help achieve it?

**Comment by ValueRaider** [comment]
> > Have you tried simply not installing `lxml`, see what works/breaks? https://github.com/ranaroussi/yfinance/discussions/1080

> Yes, yfinance and a couple of other modules won't work as they need the 4.9.2 or later version of lxml. Latest Web scraping tools.

Clearly you did not. I just uninstalled `lxml`, `yfinance` price fetches work fine.

**Comment by ValueRaider** [comment]
xda-developers.com, google.com.

**Comment by pydroid888** [comment]
#Valueraider - you have misunderstood.

Yfinance needs updated version installing after yahoo placed an api key on it.
Updated yfinance needs latest lxml version on pydroid3 else it won't install.
Latest lxml version needs to be built for pydroid3 from the relevant libraries.
Failure to have latest lxml means no upgrade to yfinace is possible = no api version of yfinance = no downloads.

Removing lxml and attempting to install latest yfinancecresults in the same error. It states clearly yfinance needs lxml >=4.9.1.

Building the latest lxml for pydroid3 requires the relevant modules to be compiled etc etc.

**Comment by ValueRaider** [comment]
I understood perfectly.

> Building the latest lxml for pydroid3 requires the relevant modules to be compiled etc etc.

Nothing to do with yfinance, we aren't Android build experts.

**Comment by pydroid888** [comment]
You obviously didn't understand and this was/is a question concerning lxml upgrades on pydroid3 necessary for yfinance to operate on that platform.

Rather than stating its nothing to do with yfinance, can you assist or not?
I gather not as you don't know how?

**Comment by pydroid888** [comment]
Since there are 100M++ Android based platforms in the world I also gather yfinance wasn't tested on any of them? A large market to ignore and push towards Aphavantage and others.

**Comment by ValueRaider** [comment]
You were specifically asking how to build lxml on Android - this is the wrong place to request Android expertise.

It *is* possible to use `yfinance` without installing lxml but then can only request price data, which is the current behaviour anyway.

**Comment by pydroid888** [comment]
No, the newer versions of yfinance WILL NOT INSTALL on pydroid3 without lxml >=4.9.1 on the platform.
It will not download price data etc. Users on an Android platform using pydroid3 will not be able to use yfinance.
It's Alphavantage or similar now. I've tried and spoken to others with the same problem.

If yfinance could operate with lxml 4.6.3 it wshould work OK.

**Comment by ValueRaider** [comment]
> No, the newer versions of yfinance WILL NOT INSTALL on pydroid3 without lxml >=4.9.1 on the platform.

I know. That's why earlier I mentioned to download yfinance directly from GitHub https://github.com/ranaroussi/yfinance/issues/1419#issuecomment-1507606414

**Comment by mritortousa** [comment]
i never got it to work on pydroid either very annoying

**Comment by pydroid888** [comment]
How can the yfinance python code on github be pip'ed and can it have the necessary lxml version be wound back on an installation towards 4.6.2? Doing that might overcome the problem otherwise yfinance is toast on the vest Python IDE for Android.

**Comment by pydroid888** [comment]
@mritortousa - have it working now. No guarantees it won't break.

1) download the source code as and unpack into a folder in root internal storage. Should unpack in yfinance-0.2.18 folder.

2) go through every file and wherever you see lmxl change version 4.9.1 to 4.6.3 and save each file that references lxml. This rolls back the lxml version needed to that available in pydroid.

3) pip install cryptography

4) pip install ./yfinance-0.2.18

See what happens and it will tell you if you missed an lxml rollback in a file but not which file. There are about 4 files to edit.

So far it works, would be good if the source allowed latest and older lxml as options.

Good luck.

I love yfinance but a real pain when it won't operate on the most popular Android Python package.

**Comment by pydroid888** [comment]
I'm not an expert coder so if I can do this others should be able to also. Learnt a fair amount via experimentation.

**Comment by mritortousa** [comment]
that script was on my pc and i still get errors . for pydroid it never works

**Comment by mritortousa** [comment]
i can't  even get the install to work on pydroid if i remember correctly are you saying that it should work on pydroid?

**Comment by mritortousa** [comment]
i try install it using pip on pydroid

i get a libxml2 and libxslt development are not installed setup egg_info did not run correctly

**Comment by pydroid888** [comment]
On your Android download the source from link below and unzip etc and follow the steps given above. So far works well. Not sure what problem there may be on windows, perhaps update pip etc and pin install cryptography?

The solution is tonroll-back the version of lxml it calls then pip install the edited source so it uses lxml 4.6.3

https://pypi.org/project/yfinance/#files

**Comment by ValueRaider** [comment]
Potentially could downgrade to 4.6.5, would that work? Anything older has serious security vulnerabilities https://security.snyk.io/package/pip/lxml . This why the min version was raised #1211

**Comment by mritortousa** [comment]
> On your Android download the source from link below and unzip etc and follow the steps given above. So far works well. Not sure what problem there may be on windows, perhaps update pip etc and pin install cryptography?
> 
> The solution is tonroll-back the version of lxml it calls then pip install the edited source so it uses lxml 4.6.3
> 
> https://pypi.org/project/yfinance/#files

do i just unzip it on my phone i am used to using pip to install it

**Comment by mritortousa** [comment]
which files do i use

[yfinance-0.2.18.tar.gz ](https://files.pythonhosted.org/packages/5b/51/364388c9b07b3df7f40487bb9dfd137521b1e1597dba31365eb6c19e0f9b/yfinance-0.2.18.tar.gz)(58.3 kB [view hashes](https://pypi.org/project/yfinance/#copy-hash-modal-480a555d-ddee-4cd5-bda6-50d44c29d53b))
Uploaded Apr 16, 2023 source

Built Distribution
[yfinance-0.2.18-py2.py3-none-any.whl ](https://files.pythonhosted.org/packages/20/e7/e8f1470e28f0532c49c37618bd141eaca5c7b99f6af23fd3794d9ef1d537/yfinance-0.2.18-py2.py3-none-any.whl)(60.3 kB [view hashes](https://pypi.org/project/yfinance/#copy-hash-modal-4eb61dc2-9285-4b91-85e8-0883b1b73007))
Uploaded Apr 16, 2023 py2 py3

**Comment by ValueRaider** [comment]
@mritortousa Read #1080

**Comment by pydroid888** [comment]
4.6.3 is the version we can use for lxml, the developer doesn't respond when asked about an update for the pydroid version, they are in Russia. A real shame as Pydroid + yfinance are a great combo on a tablet.

**Comment by pydroid888** [comment]
@mrito, its the tar.gz version, unzip and edit the source then pip install pointing to the folder where the source sits as per instructions above. Hope it works for you.

Much happier now my Samsung Tablet and yfinance play nice together with Pydroid. Can learn on the run nearly anytime.

**Comment by ValueRaider** [comment]
Why not use alternative like Termux?

**Comment by pydroid888** [comment]
It's too clunky, pydroid is much simpler to get up and running and has an intuitive i/f with useful examples and easy management. If it was better maintained it would be close to perfec for the task.

**Comment by mritortousa** [comment]
is there an easier way to get python running on android phones

**Comment by pydroid888** [comment]
pip install ./yfinance-0.2.18

Step 4 of the 4 stages above. Don't forget to edit all the lxml references first and save the edited files.

Step 4 assumes the unzipped file is in root folder of internal storage. If in another folder then use that path name to point to it.

Also, worth pip install cryptography first else it might not be there and creates an error.

**Comment by wolfegit** [comment]
My solution: From Google Play Store, update to:
Pydroid 3
Version 7.00_arm64
Updated November 8 2023

NOTE: After the upgrade, the Pydroid Respiratory Plugin need to be upgraded too.

Also, I had to reinstall all of my libraries again.  Oh, bother.

Using Pydroid's "built-in" PIP manager, I then installed these with no issue:

pandas
numpy
lxml
scikitlearn
scipy

Again, from the built-in PIP manager, I did a PIP search for "yfinance"; it was available in the search list and installed without issue.

I ran a few of my YFinance programs in Pydroid3 and, lo and behold, they worked perfectly!

Hope this helps somebody - good luck!

Sample Python code:

``` lang-python
import yfinance as yf
print("YF version:", yf.__version__)
df = yf.download("AAPL", interval="1m", start="2023-11-10", end="2023-11-11")
print("YFinance: downloaded rows:", len(df))
print(df.tail(3))
```
